WIP-SQL/group-by.md

```diff
@@ -0,0 +1,115 @@
+# GROUP BY
+author: SebaRaba
+
+levels:
+
+  - beginner
+
+  - basic
+
+type: normal
+
+category: must-know
+
+links:
+
+  - '[More on GROUP BY](https://www.techonthenet.com/sql/group_by.php)'
+
+---
+## Content
+
+The `GROUP BY` clause is often used with aggregate functions (e.g. `SUM`, `AVG`). Syntactically it comes after the `WHERE` clause and before the `ORDER BY` in the *query*. It enables users to group data by one or more columns.
+
+In our Pokemon db, we have a table called *move_name* with the following cloumns:
+- id - unique move name id
+- language_id - id of the language name is spelled in
+- move_id
+- name - how each move is called in each language
+`GROUP BY` will help us count how many move_names are there for each langauge:
+
+```
+SELECT COUNT(name), language_id
+FROM move_name
+GROUP BY language_id;
+
+--Result:
+count | language_id
+------+-------------
+  639 |           6
+  639 |           5
+  639 |           8
+  639 |           1
+  639 |           9
+  617 |           3
+  639 |           7
+(7 rows)
+
+```
+---
+## Practice
+
+We have a table called pokemon. Two of its fields are:
+- is_default - it can be either true or false
+- name
+We want to know how many pokemons are (not) default:
+```
+SELECT ???(name),???
+FROM pokemon
+??? is_default;
+
+--Result:
+count | is_default
+------+------------
+   90 | f
+  721 | t
+(2 rows)
+```
+
+* COUNT
+* is_default
+* GROUP BY
+* AVG
+* SUM
+* name
+
+---
+## Revision
+
+In our pokemon db item is a table with the following columns:
+- id - unique id of the item
+- cost - item's cost
+- fling_power - "Dark type move"[1]
+- item_category_id
+- item_fling_effect_id
+- name - item's name
+We would like to get the average cost of each item category.
+
+Note that there are a total of 44 categories in our db. But for space and readability purposes we will display the result for the first 10 categories.
```

Comment:
you can always put a 
...
under the table to denote "this continues, but is cut off", then you don't have to have all this extra text.